src/main/kotlin/frc/team5499/frc2018Kotlin/path/Path.kt

```diff
@@ -0,0 +1,100 @@
+package frc.team5499.frc2018Kotlin.path
+
+import java.io.FileNotFoundException
+import java.io.FileReader
+
+import com.opencsv.CSVReader
+
+import frc.team5499.frc2018Kotlin.Constants
+import frc.team5499.frc2018Kotlin.utils.Vector2
+
+public class Path(filepath: String, backwards: Boolean = false) {
+
+    private var coordinates: MutableList<Vector2> = mutableListOf<Vector2>()
+    private var targetVelocities: MutableList<Double> = mutableListOf<Double>()
+
+    private var backwards: Boolean = false
+
+    init {
+        this.backwards = backwards
+
+        var tempCoords: MutableList<Vector2> = mutableListOf<Vector2>()
+        var tempVelo: MutableList<Double> = mutableListOf<Double>()
+        try {
+            var reader: CSVReader = CSVReader(FileReader(filepath))
+            var line: Array<String> = reader.readNext()
+            while (line != null) {
+                tempCoords.add(Vector2(line[0].toDouble(), line[1].toDouble()))
+                if (backwards) { tempVelo.add(-line[2].toDouble()) } else { tempVelo.add(line[2].toDouble()) }
+                line = reader.readNext()
+            }
+            reader.close()
+
+            for (i in tempCoords.indices) {
+                coordinates[i] = tempCoords.get(i)
+                targetVelocities[i] = tempVelo.get(i)
+            }
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
+            System.exit(1)

```

Comment:
Do we exit in the java code? I dont know If we should do that

Reply:
Since this likely going to be used for autonomous, it makes since to keep running and fall back to a default mode. I think we should let the calling function catch the exception and decide what to do with it.

Reply:
yeah, it should exit. If the path can't be found, that is a fatal error. Causing the program to exit will force the programmer to resolve the issue before continuing.

Reply:
no, if it exits in auto, then we cant run in teleop either until the robot reboots.

Reply:
I agree with Connor

Reply:
hmm

Reply:
What 254 does is they package all the paths with the jar file. This is probably a better idea, because then the programmer is forced to update the paths locally, as well as catch any errors before the code will work. You will be less likely to run into issues with the paths being out of date or different versions on the robot.

Reply:
how do you package the paths with the jar?

Reply:
I know their 2017 code used actual java classes to represent paths